### DISCUSSION TITLE
how to print keys and values of static_map , can not find accessed function to do it ?

### DISCUSSION THREAD

**Discussion by bug-code** [NVIDIA/cuCollections#372] [Q&A] [answered]
as the topic title

**Comment by sleeepyjack** [comment, marked as answer]
Here you go: https://godbolt.org/z/Kvn9hPcPo

I just altered the `examples/static_map/host_bulk_example.cu` to do what you want to achieve.

I'm assuming performance/memory footprint isn't of concern, as printing out the actual key/value pairs is mostly used for debugging purposes. Thus, we can just compact all the key/value pairs stored in the map into a new vector.
Also, I'm assuming you're using the `cuco::static_map` and not the new `cuco::experimental::static_map`? There's only a slight API difference between the two.

**Reply by bug-code** [reply]
Thanks, this helps a lot
